corehq/apps/app_manager/urls.py

```diff
@@ -105,9 +105,9 @@
     url(r'^languages/bulk_app_translations/download/$', download_bulk_app_translations, name='download_bulk_app_translations'),
     url(r'^languages/bulk_app_translations/upload/$', upload_bulk_app_translations, name='upload_bulk_app_translations'),
     url(r'^multimedia_ajax/$', multimedia_ajax, name='app_multimedia_ajax'),
-    url(r'^multimedia_sizes/$', get_multimedia_sizes, name='get_multimedia_sizes'),
-    url(r'^compare_multimedia_sizes/(?P<other_build_id>[\w-]+)/$', compare_multimedia_sizes,
-        name='compare_multimedia_sizes'),
+    url(r'^multimedia_sizes/(?P<build_profile_id>[\w-]+)/$', get_multimedia_sizes, name='get_multimedia_sizes'),
```

Comment:
If you left both the old URLs and new ones, wouldn't the old one just call `get_multimedia_sizes` with `build_profile_id` defaulting to `None`? That would let you get rid of the magic string logic (`build_profile_id == "default"`) which is a code smell.

Reply:
ya that "default" business is not a great way out but the only i could think of.
```
    url(r'^multimedia_sizes/$', get_multimedia_sizes, name='get_multimedia_sizes_without_build_profile'),
    url(r'^multimedia_sizes/(?P<build_profile_id>[\w-]+)/$', get_multimedia_sizes, name='get_multimedia_sizes'),
```
thinking about other changes this would need. Seeing following for just getting mm sizes without build profile id
```
corehq/apps/app_manager/templates/app_manager/source_files.html

-  {% registerurl 'get_multimedia_sizes' app.domain '---' '---' %}
+  {% registerurl 'get_multimedia_sizes' app.domain '---' %}
+  {% registerurl 'get_multimedia_sizes_for_build_profile' app.domain '---' '---' %}

corehq/apps/app_manager/urls.py

-    url(r'^multimedia_sizes/(?P<build_profile_id>[\w-]+)/$', get_multimedia_sizes, name='get_multimedia_sizes'),
+    url(r'^multimedia_sizes/$', get_multimedia_sizes, name='get_multimedia_sizes'),
+    url(r'^multimedia_sizes/(?P<build_profile_id>[\w-]+)/$', get_multimedia_sizes,
+        name='get_multimedia_sizes_for_build_profile'),

corehq/apps/app_manager/views/multimedia.py
 @require_deploy_apps
 @quickcache(['domain', 'app_id', 'build_profile_id'], timeout=60 * 60)
-def get_multimedia_sizes(request, domain, app_id, build_profile_id):
+def get_multimedia_sizes(request, domain, app_id, build_profile_id=None):
```
i think its about the cost of handling in python vs handling in JS and JS sounds better to me. Thoughts?

Reply:
It would be a bit of code to add, yes, but that code would be more standard/boring which is good. `get_multimedia_sizes_for_build` struck me as especially unusual because it defaults the `build_profile_id` param to `None` but it also has to check if it's equal to "default"

Reply:
the "default" check won't be there. I was just seeing the changes i would need to make to get the url as needed first.
going ahead to make these changes then

Reply:
https://github.com/dimagi/commcare-hq/pull/26032/commits/1c30039682ec1158ce087335c8d6e95dd6bb9928